flight/targets/sparky2/fw/pios_board.c

```diff
@@ -696,8 +699,20 @@ void PIOS_Board_Init(void) {
 	case HWSPARKY2_ADCDAC_FSKTELEM:
 #if defined(PIOS_INCLUDE_FSK)
 	{
+		uintptr_t fskdac_id;
+		PIOS_FSKDAC_Init(&fskdac_id);
+
 		uintptr_t fskdac_com_id;
-		PIOS_FSKDAC_Init(&fskdac_com_id, &pios_fskdac_config);
+		uint8_t * tx_buffer = (uint8_t *) PIOS_malloc(PIOS_COM_FSKDAC_BUF_LEN);
+		PIOS_Assert(tx_buffer);
+		if (PIOS_COM_Init(&fskdac_com_id, &pios_fskdac_com_driver, fskdac_id,
+		                  NULL, 0,
+		                  tx_buffer, PIOS_COM_FSKDAC_BUF_LEN))
+			panic(6);
```

Comment:
Cool. Looking forward to it.

On a related note, I've been thinking about going through the entire firmware codebase and fixing all the errors returned by all the functions. Maybe just bundle them all into one giant enum. I think it would help us do better error handling if we had names for them, instead of just `-1`, `0`, etc...